### ISSUE TITLE
Download as JSON Button for Runbook Process fails due to incorrect api call

### ISSUE THREAD

**Issue by millerjn21** [OctopusDeploy/Issues#6515]
# Prerequisites

- [X] I have verified the problem exists in the latest version
- [X] I have searched [open](https://github.com/OctopusDeploy/Issues/issues) and [closed](https://github.com/OctopusDeploy/Issues/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed) issues to make sure it isn't already reported
- [X] I have written a descriptive issue title
- [X] I have linked the original source of this report
- [X] I have tagged the issue appropriately (area/*, kind/bug, tag/regression?)

# The bug

In 2020.3.2, the Download as JSON button in the Runbook Process section points to the wrong api endpoint. It points to

 api/space/**deploymentprocesses**/runbookprocess-runbookId

This causes the api call to 404 and the download wont work.

## What I expected to happen

I expect the api call to go to the following endpoint and succeed.

api/space/**runbookprocesses**/runbookprocess-runbookID

## Steps to reproduce

1. Go to a runbook
2. Go to the process section
3. Click the overflow in the upper right
4. Click Download as JSON
5. Nothing downloads due to bad api call

### Screen capture

![image](https://user-images.githubusercontent.com/63259411/90139641-a4c45180-dd46-11ea-99c7-506b29341ce1.png)

### Log exerpt

```
N/A

```

## Affected versions

**Octopus Server:** 2020.3.2. I assume it's broken on anything 2020.3.X, it works on 2020.2.16.

## Workarounds

Manually make the API call to get the JSON with this code I wrote:

```
$OctopusServerUrl = ""  #PUT YOUR SERVER LOCATION HERE. (e.g. http://localhost)
$ApiKey = ""   #PUT YOUR API KEY HERE
$spaceName = ""   #PUT SPACE NAME HERE
$projectName = "" #PUT PROJECT NAME HERE
$runbookName = "" #PUT RUNBOOK NAME HERE

Function Get-SpaceId {
    # Define parameters
    param(
        $Space
    )
    $spaceName = $Space
    $spaceList = Invoke-RestMethod "$OctopusServerUrl/api/spaces?Name=$spaceName" -Headers @{"X-Octopus-ApiKey" = $ApiKey }
    $spaceFilter = @($spaceList.Items | Where { $_.Name -eq $spaceName })
    $spaceId = $spaceFilter[0].Id
    return $spaceId
}
Function Get-OctopusProject {
    # Define parameters
    param(
        $OctopusServerUrl,
        $ApiKey,
        $ProjectName,
        $SpaceId
    )
    # Call API to get all projects, then filter on name
    $octopusProject = Invoke-RestMethod -Method "get" -Uri "$OctopusServerUrl/api/projects/all" -Headers @{"X-Octopus-ApiKey" = "$ApiKey" }

    # return the specific project
    return ($octopusProject | Where-Object { $_.Name -eq $ProjectName -and $_.SpaceId -eq $SpaceId})
}
Function Get-RunbookId {
    # Define parameters
    param(
        $Runbook,
        $ProjectId
    )
    $runbookname = $runbook
    $runbookList = Invoke-RestMethod "$OctopusServerUrl/api/runbooks?Name=$runbookName" -Headers @{"X-Octopus-ApiKey" = $ApiKey }
    $runbookFilter = @($runbookList.Items | Where { $_.Name -eq $runbookName -and $_.ProjectId -eq $ProjectId })
    $runbookId = $runbookFilter[0].Id
    return $runbookId
}


$spaceId = Get-SpaceId -space $spaceName
$projectId = Get-OctopusProject -OctopusServerUrl $OctopusServerUrl -ApiKey $ApiKey -ProjectName $projectName -SpaceId $spaceid
$projectId = $projectId.Id
$runbookId = Get-RunbookId -Runbook $runbookName -ProjectId $projectId
$runbook = Invoke-RestMethod "$OctopusServerUrl/api/$spaceId/runbookProcesses/RunbookProcess-$runbookId" -Headers @{"X-Octopus-ApiKey" = $ApiKey }
$json = $null
$json = $runbook | convertto-json -depth 10
write-host $json
```

## Links

https://help.octopus.com/t/runbook-download-as-json-generates-error/25530/3

**Comment by mcasperson** [comment]
We have flagged this as an issue with product, but at this point we do not have capacity to assign an engineer to resolve the issue. We'll keep the issue open, but unfortunately we can not provide any guidance as to when it will be resolved.

**Comment by millerjn21** [comment]
As matt stated above, this is fixed in 2020.3.3. I tested it. Thank you for the heads up.

**Comment by matt-richardson** [comment]
Release Note: Fixed the Download as JSON menu item for runbooks
